pkg/xray/resource_xray_repository_config.go

```diff
@@ -0,0 +1,349 @@
+package xray
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jfrog/terraform-provider-shared/util"
+	"github.com/jfrog/terraform-provider-shared/validator"
+)
+
+type GeneralRepoConfig struct {
+	// Omitempty is used because 'vuln_contextual_analysis' is not supported by self-hosted Xray installation.
+	VulnContextualAnalysis bool `json:"vuln_contextual_analysis,omitempty"`
+	RetentionInDays        int  `json:"retention_in_days,omitempty"`
+}
+
+type RepoPathsConfiguration struct {
+	PatternList    []Pattern         `json:"patterns,omitempty"`
+	OtherArtifacts AllOtherArtifacts `json:"all_other_artifacts,omitempty"`
+}
+
+type Pattern struct {
+	Include           string `json:"include"`
+	Exclude           string `json:"exclude"`
+	IndexNewArtifacts bool   `json:"index_new_artifacts"`
+	RetentionInDays   int    `json:"retention_in_days"`
+}
+
+type AllOtherArtifacts struct {
+	IndexNewArtifacts bool `json:"index_new_artifacts"`
+	RetentionInDays   int  `json:"retention_in_days"`
+}
+
+type RepositoryConfiguration struct {
+	RepoName string `json:"repo_name"`
+	// Pointer is used to be able to verify if the GeneralRepoConfig or RepoPathsConfiguration struct is nil
+	RepoConfig      *GeneralRepoConfig      `json:"repo_config,omitempty"`
+	RepoPathsConfig *RepoPathsConfiguration `json:"repo_paths_config,omitempty"`
+}
+
+func resourceXrayRepositoryConfig() *schema.Resource {
+	var repositoryConfigSchema = util.MergeMaps(
+		map[string]*schema.Schema{
+			"repo_name": {
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      `Repository name.`,
+				ValidateDiagFunc: validator.StringIsNotEmpty,
+			},
+			"repo_config": {
+				Type:          schema.TypeSet,
+				Optional:      true,
+				Description:   `Single repository configuration. Only one of 'repo_config' or 'repo_paths_config' can be set.`,
+				ConflictsWith: []string{"repo_paths_config"},
```

Comment:
We can have one for self-hosted installations, but for SaaS we can have 2 in the next Xray version. I'll keep it as is.

Reply:
You're right, I misunderstood Min and Max. Also, in this case `MinItems` is pointless to set, because:
```
	// If the field Optional is set to true then MinItems is ignored and thus
	// effectively zero.